File: 13. Roman to Integer/solution.java
class Solution {
  public int romanToInt(String s) {
    int index = 0;
    int result = 0;
    Map<String, Integer> dictionary = new HashMap<String, Integer>(){{
      put("I", 1);
      put("IV", 4);
      put("V", 5);
      put("IX", 9);
      put("X", 10);
      put("XL", 40);
      put("L", 50);
      put("XC", 90);
      put("C", 100);
      put("CD", 400);
      put("D", 500);
      put("CM", 900);
      put("M", 1000);
    }};
    while (index < s.length()){
      if (index + 1 < s.length() && dictionary.containsKey("" + s.charAt(index) + s.charAt(index + 1))){
        result += dictionary.get("" + s.charAt(index) + s.charAt(index + 1));
        index += 2;
      }
      else{
        result += dictionary.get("" + s.charAt(index));
        index += 1;
      }
    }
    return result;
  }
}

File: 13. Roman to Integer/solution.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(s) {
  let index = 0;
  let result = 0;
  let dictionary = {
    "I": 1,
    "IV": 4,
    "V": 5,
    "IX": 9,
    "X": 10,
    "XL": 40,
    "L": 50,
    "XC": 90,
    "C": 100,
    "CD": 400,
    "D": 500,
    "CM": 900,
    "M": 1000
  };
  while (index < s.length){
    if (index + 1 < s.length && (dictionary[s[index] + s[index + 1]])){
      result += dictionary[s[index] + s[index + 1]];
      index += 2;
    }
    else{
      result += dictionary[s[index]];
      index += 1;
    }
  }
  return result;
};